### PR TITLE
Spring 07

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dto/request/MovieSessionRequestDto.java [hunk 1]
```diff
@@ -1,8 +1,13 @@
 package com.mate.dto.request;
 
+import javax.validation.constraints.NotNull;
+
 public class MovieSessionRequestDto {
+    @NotNull
     private Long movieId;
```
Comment (tonitrus55) [inline src/main/java/com/mate/dto/request/MovieSessionRequestDto.java:8]: Что если тебе прийдет -1?

File: src/main/java/com/mate/dto/request/MovieSessionRequestDto.java [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
+    @NotNull
     private Long cinemaHallId;
```
Comment (tonitrus55) [inline src/main/java/com/mate/dto/request/MovieSessionRequestDto.java:10]: Same

File: src/main/java/com/mate/lib/EmailValidator.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    public static final String EMAIL_VALIDATION_REGEX = "^(.+)@(.+)$";
+
+    @Override
+    public void initialize(ValidEmail constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String field, ConstraintValidatorContext context) {
+        return field.matches(EMAIL_VALIDATION_REGEX);
```
Comment (tonitrus55) [inline src/main/java/com/mate/lib/EmailValidator.java]: Remember, that null.equals(smth) == NPE. HINT: be careful with implementation of the isValid() method.

**Review by Alekearl — APPROVED** [review]

**Review by tonitrus55 — APPROVED** [review]
GJ

File: src/main/java/com/mate/dto/request/UserRequestDto.java [hunk 4]
```diff
@@ -11,4 +23,8 @@ public String getEmail() {
     public String getPassword() {
         return password;
     }
+
+    public String getRepeatPassword() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/dto/request/UserRequestDto.java:27]: where is the setter?

File: src/main/java/com/mate/lib/EmailValidator.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.mate.lib;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/lib/EmailValidator.java]: redundant empty line

File: src/main/java/com/mate/lib/EmailValidator.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    public static final String EMAIL_VALIDATION_REGEX = "^(.+)@(.+)$";
+
+    @Override
+    public void initialize(ValidEmail constraintAnnotation) {
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/lib/EmailValidator.java]: we can remove this

File: src/main/java/com/mate/lib/FieldsValueMatch.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+@Constraint(validatedBy = FieldsValueMatchValidator.class)
+@Target({ ElementType.TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+public @interface FieldsValueMatch {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/lib/FieldsValueMatch.java]: redundant empty line

File: src/main/java/com/mate/lib/FieldsValueMatchValidator.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package com.mate.lib;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldsValueMatchValidator
+        implements ConstraintValidator<FieldsValueMatch, Object> {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/lib/FieldsValueMatchValidator.java]: same

File: src/main/java/com/mate/lib/FieldsValueMatchValidator.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    private String fieldMatch;
+
+    public void initialize(FieldsValueMatch constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext context) {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/mate/lib/FieldsValueMatchValidator.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**